modules/shared/src/main/scala/org/tessellation/currency/validations/FeeTransactionValidations.scala

```diff
@@ -0,0 +1,88 @@
+package org.tessellation.currency.validations
+
+import cats.data.{NonEmptyList, ValidatedNec}
+import cats.effect.Async
+import cats.syntax.all._
+
+import org.tessellation.currency.dataApplication._
+import org.tessellation.schema.address.Address
+import org.tessellation.schema.balance.Balance
+import org.tessellation.security.SecurityProvider
+import org.tessellation.security.hash.Hash
+import org.tessellation.security.signature.Signed
+
+object FeeTransactionValidations {
+
+  private def validateFeeTransactionHashMatch[F[_]: Async](
+    feeTransaction: Signed[FeeTransaction],
+    updates: NonEmptyList[Signed[DataTransaction]],
+    dataApplication: BaseDataApplicationService[F]
+  ): F[ValidatedNec[DataApplicationValidationError, Unit]] =
+    updates.existsM {
+      case Signed(dataUpdate: DataUpdate, _) =>
+        dataApplication.serializeUpdate(dataUpdate).map { serializedUpdate =>
+          Hash.fromBytes(serializedUpdate) === feeTransaction.dataUpdateRef
```

Comment:
previously I've seen that you wrapped it into `.delay`. so that delay was not necessary?

Reply:
Yeah, I'll update